ncm-systemd/src/main/resources/regular/tests/profiles/unitfile.pan

```diff
@@ -52,6 +52,9 @@ bind "/unitfile" = systemd_unitfile_config[];
         'TTYVHangup', false,
         'LimitSTACK', -1,
         'LimitNPROC', 100,
+        'RuntimeDirectory', list('foo/bar', 'tmp'),
+        'RuntimeDirectoryMode', 0777,
```

Comment:
Interesting - so 0777 is a valid long in PAN?!  Not sure that's obvious from someone looking at the schema.

Reply:
I'm not sure I understand your remark about the schema?

I've tested it empirically and 0777 produces 777 in the output. If I remove the filter part, it becomes 511. So it seems like pan knows and reads 0777 as octal.

Reply:
@ned21 pan supports octal (any long that starts with an explicit `0`) and hex (same, but start with `0x`)
it's not obvious from the schema, since there's nothing to it. perhaps the docstring might use an example.

Reply:
even in docs https://quattor-pan.readthedocs.io/en/stable/pan-book/pan-book.html#properties-and-primitive-types

Reply:
Yep, sorry I had a nagging doubt about this earlier. I remember there was a nasty regression in an one version of filecopy due to this not being handled well and I wanted a bit more time to think about it. 

I think there are two issues:
1. As a user should I specify 0777 or 777? Do they do the same thing? The answer is that they do mean different things in PAN (which can lead to subtle bugs) so it's down the implementor to make sure that the user intent is converted correctly.
2. 777 and 1777 are valid decimal numbers which are also valid modes when interpreted as a string. What happens to more esoteric modes like 011 and 0001? Again if the implementation is sufficiently robust then the right thing will happen.

So based on the unit test, it looks like the code is good. How do we communicate to a user that either form is acceptable? I don't think we need to spell it out in every line of the schema, but it might be useful to mention it somewhere, perhaps by way of example as was suggested.

Reply:
I can add a docstring? The output is in octal. So you can write, 0777, 511 or 0x1FF to get 777. 
However, write 777 and you will get 1411. 

If you want both 777 and 0777 to produce the same thing, I need to use a string. 

What's the preferred way?

Reply:
there has to be a limit on the size to start `0..511` or `0..1023` if you want sticky bits.
but this indeed can be confusing for endusers. `0511` and `511` are valid and very different.
if you want to protect against it, you need to add compiler support for it. otherwise a string that parses as an octal is the only option.

Reply:
> However, write 777 and you will get 1411.

yeah, so IMHO that's not user friendly. And if the number needs to always be written in octal then does 1777 actually need to be written as 01777? 

In which case I think I agree that using a string makes more sense here.

Reply:
@ned21 I've pushed a commit that uses a string (with regex) for the modes/masks.